src/datumaro/components/annotation.py

```diff
@@ -501,6 +503,41 @@ def __eq__(self, other):
         return self.rle == other.rle
 
 
+@attrs(slots=True, eq=False, order=False)
+class ExtractedMask(Mask):
+    """Mask annotation (binary mask) extracted from an index mask (integer 2D Numpy array).
+
+    This class can extract a binary mask with given index mask and index value.
+    The advantage of this class is that we can create multiple binary mask but they share a single index mask source.
+
+    Attributes:
+        index_mask: Integer 2D Numpy array. Its pixel can indicate a label id (class) or an instance id.
+        index: Integer value to extract a binary mask from the given index mask.
+
+    Examples:
+        >>> import numpy as np
+        >>> from datumaro.components.annotation import ExtractedMask
+        >>>
+        >>> index_mask = np.random.randint(low=1, high=3, size=(10, 10), dtype=np.uint8)
+        >>> mask1 = ExtractedMask(index_mask=index_mask, index=1, label=1) #
+        >>> mask2 = ExtractedMask(index_mask=index_mask, index=2, label=2) #
+        >>> np.unique(mask1.image).tolist()  # `image` property create a binary mask
+        np.array([0, 1])
+        >>> mask1.index_mask == mask2.index_mask  # They share the same source
+        True
+    """
+
+    index_mask: Union[IndexMaskImage, IndexMaskImageCallable] = field()
```

Comment:
This looks like `instance_mask` for me. How about using `instance_mask` instead?

Reply:
Likewise https://github.com/openvinotoolkit/datumaro/pull/1480#discussion_r1582527296, `index_mask` is 2d integer map which pixel can indicate an instance id or label id. It is available to represent both.

Reply:
So I am saying about the rename of `index_mask` to `instance_mask` for keeping the consistencies between Datumaro.

Reply:
I thought that there is two mask types broadly accepted in the ML community: 1) instance mask and 2) class mask. The instance mask represents an unique identifier for object. For example, there are two apples, but they can have two instance mask values such as 0 for the first apple and 1 for the second apple. On the other hand, class mask represents a semantic label. In the previously example, both apples will have 1 class mask values (0 for background and 1 for apple).

However, this time, I introduced a third terminology which can express both. It is possible that we can design `ExtractedMask` to specify the source mask one of either such as
```python
class ExtractedMask:
     source: np.ndarray
     source_type: Literal["instance", "class"]
```
, and it was my first thought but I changed this design to the current one. The reason why I changed my mind can be found in this comment, https://github.com/openvinotoolkit/datumaro/pull/1480/files#r1582581086. This data format tries to import something that is not clear whether its source is instance mask or class mask. I thought that it can be happened by any chance, so I'd like to introduce more general abstraction for this by introducing `index_mask` and `index`.